File: examples/java/src/main/java/com/exemplo/Insecure.java
package com.exemplo;

import java.io.IOException;

/**
 * VULNERABILIDADE PLANTADA — NÃO CORRIJA.
 *
 * <p>Passar entrada do usuário direto para {@code Runtime.exec} é command injection.
 * O Semgrep reporta isto pelos rulesets p/security-audit e p/owasp-top-ten.
 *
 * <p>É este achado que o self-test da plataforma usa para provar que o SAST está realmente
 * analisando o código e que o gate de severidade funciona. Remover isto faz o job
 * {@code sast-detecta-vulnerabilidade} do _selftest.yml falhar.
 */
public final class Insecure {

    private Insecure() {
    }

    /** Executa um comando vindo do usuário. Deliberadamente inseguro. */
    public static Process listarArquivos(String diretorioDoUsuario) throws IOException {
        return Runtime.getRuntime().exec("ls " + diretorioDoUsuario);
    }
}

File: examples/go/insecure.go
package main

import (
	"crypto/md5" // #nosec G501 -- intencional: ver comentário abaixo
	"encoding/hex"
)

// VULNERABILIDADE PLANTADA — NÃO CORRIJA.
//
// MD5 é criptograficamente quebrado. O gosec reporta isto como G401 e o Semgrep também
// o detecta. É este achado que o self-test da plataforma usa para provar que o SAST
// está realmente analisando o código e que o gate de severidade funciona.
// Remover isto faz o teste `sast-detecta-vulnerabilidade` do _selftest.yml falhar.
func WeakHash(data string) string {
	h := md5.New() // #nosec G401
	h.Write([]byte(data))
	return hex.EncodeToString(h.Sum(nil))
}
